MovieAppPetProject/MovieAppPetProject/HomeScreen/NewHomeViewController.swift

```diff
@@ -0,0 +1,103 @@
+                return UITableViewCell()
+            }
+            cell.configure(with: nowPlayingViewModel.nowPlayingMovies ?? [], sectionTitle: " Now Playing")
+            cell.didSelectItem = { [weak self] movie in
+                guard let self = self else { return }
+                self.performSegue(withIdentifier: Constants.Identifiers.goToMovieDetails, sender: movie.movieID)
+            }
+            return cell
+        case 1:
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Identifiers.homeScreenTableCell, for: indexPath) as? NewTableViewCell else {
+                return UITableViewCell()
+            }
+            cell.configure(with: popularMoviesViewModel.popularMovies ?? [], sectionTitle: " Popular")
+            cell.didSelectItem = { [weak self] movie in
+                guard let self = self else { return }
+                self.performSegue(withIdentifier: Constants.Identifiers.goToMovieDetails, sender: movie.movieID)
+            }
+            return cell
+        case 2:
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Identifiers.homeScreenTableCell, for: indexPath) as? NewTableViewCell else {
+                return UITableViewCell()
+            }
+            cell.configure(with: topRatedMoviesViewModel.topRatedMovies ?? [], sectionTitle: " Top Rated")
+            cell.didSelectItem = { [weak self] movie in
+                guard let self = self else { return }
+                self.performSegue(withIdentifier: Constants.Identifiers.goToMovieDetails, sender: movie.movieID)
+            }
+            return cell
+        case 3:
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Identifiers.homeScreenTableCell, for: indexPath) as? NewTableViewCell else {
+                return UITableViewCell()
+            }
+            cell.configure(with: upcomingMoviesViewModel.upcomingMovies ?? [], sectionTitle: " Upcoming")
+            cell.didSelectItem = { [weak self] movie in
+                guard let self = self else { return }
+                self.performSegue(withIdentifier: Constants.Identifiers.goToMovieDetails, sender: movie.movieID)
+            }
+            return cell
+        default:
+            return UITableViewCell()
+        }
+    }
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == Constants.Identifiers.goToMovieDetails,
+            let movieDetailsVC = segue.destination as? MovieDetailsViewController,
+            let movieId = sender as? Int {
+            movieDetailsVC.setMovieID(movieID: movieId)
+        }
+    }
+}
+
+// MARK: - Delegate
+extension NewHomeViewController: NowPlayingViewModelDelegate, UpcomingMoviesViewModelDelegate, PopularMoviesViewModelDelegate, TopRatedMoviesViewModelDelegate {
```

Comment:
why does it conform to o many delegates?
Just make 1 protocol that all of them conform to then put it here.
Please also don't call them delegates. They are protocols

Reply:
change made